packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1753,6 +1753,26 @@ The findRenderObject() method was called for the following element:
     child.dependOnInheritedElement(ancestor);
     expect(child.doesDependOnInheritedElement(ancestor), isTrue);
   });
+
+  testWidgets(
+      'RenderObjectElement updateChildren modifying GlobalKey Widget index and parent',
+      (WidgetTester tester) async {
+    final GlobalKey globalKey = GlobalKey();
```

Comment:
Add a relation link
// Regression test for https://github.com/flutter/flutter/issues/120762